src/xio.c

```diff
@@ -218,7 +218,7 @@ static void* xio_CloneOption(const char* name, const void* value)
     {
         if (strcmp(name, CONCRETE_OPTIONS) == 0)
         {
-            result = (void*)value;
+            result = (void*)OptionHandler_Clone((OPTIONHANDLER_HANDLE)value);
```

Comment:
weird so before this clone would just copy the pointer?

Reply:
Yes, it was just passing the pointer instead of copying it. This is called a "shallow copy", which is not a good practice in our code.